Avoid assertion on nonblocking connect()

From: Christian Prochaska <christian.prochaska@genode-labs.com>

When calling 'connect()' in nonblocking mode and the connection has been
established, don't call 'tcp_connect()' again, which would trigger an
assertion with the message 'tcp_connect: can only connect from state
CLOSED'.
---
diff --git a/src/api/api_msg.c b/src/api/api_msg.c
index 36efa79..9d0bc10 100644
--- a/src/api/api_msg.c
+++ b/src/api/api_msg.c
@@ -1237,7 +1237,8 @@ lwip_netconn_do_connect(void *m)
       /* Prevent connect while doing any other action. */
       if (msg->conn->state == NETCONN_CONNECT) {
         msg->err = ERR_ALREADY;
-      } else if (msg->conn->state != NETCONN_NONE) {
+      } else if ((msg->conn->state != NETCONN_NONE) ||
+                 ((msg->conn->pcb.tcp->state == ESTABLISHED) && netconn_is_nonblocking(msg->conn))) {
         msg->err = ERR_ISCONN;
       } else {
         setup_tcp(msg->conn);
